crates/compilers/src/compilers/resolc/compiler.rs

```diff
@@ -0,0 +1,220 @@
+use crate::{
+    error::{Result, SolcError},
+    resolver::parse::SolData,
+    solc::{Solc, SolcCompiler, SolcSettings},
+    Compiler, CompilerVersion,
+};
+use foundry_compilers_artifacts::{resolc::ResolcCompilerOutput, Contract, Error, SolcLanguage};
+use itertools::Itertools;
+use semver::Version;
+use serde::Serialize;
+use std::{
+    io,
+    io::Write,
+    path::{Path, PathBuf},
+    process::{Command, Output, Stdio},
+    str::FromStr,
+};
+
+use super::{ResolcInput, ResolcVersionedInput};
+
+#[derive(Clone, Debug)]
+pub struct Resolc {
+    pub resolc: PathBuf,
+    pub resolc_version: Version,
+    pub solc: SolcCompiler,
+}
+
+impl Compiler for Resolc {
+    type CompilerContract = Contract;
+    type Input = ResolcVersionedInput;
+    type CompilationError = Error;
+    type ParsedSource = SolData;
+    type Settings = SolcSettings;
+    type Language = SolcLanguage;
+
+    /// Instead of using specific sols version we are going to autodetect
+    /// Installed versions
+    fn available_versions(&self, language: &SolcLanguage) -> Vec<CompilerVersion> {
+        self.solc
+            .available_versions(language)
+            .into_iter()
+            .filter(|version| match version {
+                CompilerVersion::Installed(version) | CompilerVersion::Remote(version) => {
+                    version.minor >= 8
+                }
+            })
+            .collect::<Vec<_>>()
+    }
+
+    fn compile(
+        &self,
+        input: &Self::Input,
+    ) -> Result<crate::compilers::CompilerOutput<Error, Self::CompilerContract>, SolcError> {
+        let solc = self.solc(input)?;
+        let results = self.compile_output::<ResolcInput>(&solc, &input.input)?;
+        let output = std::str::from_utf8(&results).map_err(|_| SolcError::InvalidUtf8)?;
+        let results: ResolcCompilerOutput =
+            serde_json::from_str(output).map_err(|e| SolcError::msg(e.to_string()))?;
+        Ok(results.into())
+    }
+}
+
+impl Default for Resolc {
+    fn default() -> Self {
```

Comment:
Isnt the default method always meant to pass? I see that we throwing errors here?